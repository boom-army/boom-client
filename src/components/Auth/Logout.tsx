import React, { useCallback, useContext } from "react";
import { ThemeContext } from "../../contexts/theme";
import { UserIcon } from "../Icons";
import { Wrapper } from "../ToggleTheme";
import { useSnackbar } from "notistack";
import { useWallet } from "@solana/wallet-adapter-react";

const Logout = () => {
  const { theme } = useContext(ThemeContext);
  const { disconnect } = useWallet();
  const { enqueueSnackbar } = useSnackbar();

  const handleLogout = useCallback(
    (event) => {
      localStorage.clear();
      disconnect().catch(() => {
        // Silently catch because any errors are caught by the context `onError` handler
      });
      setTimeout(() => {
        window.location.href = "/";
      }, 2100);
      window.location.reload();
      return enqueueSnackbar("You are logged out", { variant: "success" });
    },
    [disconnect, enqueueSnackbar]
  );

  return (
    <Wrapper onClick={handleLogout}>
      <UserIcon sm="true" color={theme.accentColor} />
      <p>Logout</p>
    </Wrapper>
  );
};

export default Logout;
