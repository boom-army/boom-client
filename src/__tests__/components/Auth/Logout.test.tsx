import React from "react";
import Logout from "../../../components/Auth/Logout";
import { act, fireEvent, render } from "@testing-library/react";
import { SnackbarProvider } from "notistack";

describe("<Logout/> component :", () => {
  test("display <Logout/> component ", async () => {
    let rendered = render(
      <SnackbarProvider>
        <Logout />
      </SnackbarProvider>
    );
    expect(rendered).toMatchSnapshot();
  });
});
