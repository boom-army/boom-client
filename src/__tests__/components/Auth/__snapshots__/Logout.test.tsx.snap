// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout/> component : display <Logout/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1qx74l9"
      >
        <svg
          class="css-3qrrh5"
          color="#CA2055"
          height="24"
          sm="true"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm7.753 18.305c-.261-.586-.789-.991-1.871-1.241-2.293-.529-4.428-.993-3.393-2.945 3.145-5.942.833-9.119-2.489-9.119-3.388 0-5.644 3.299-2.489 9.119 1.066 1.964-1.148 2.427-3.393 2.945-1.084.25-1.608.658-1.867 1.246-1.405-1.723-2.251-3.919-2.251-6.31 0-5.514 4.486-10 10-10s10 4.486 10 10c0 2.389-.845 4.583-2.247 6.305z"
          />
        </svg>
        <p>
          Logout
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1qx74l9"
    >
      <svg
        class="css-3qrrh5"
        color="#CA2055"
        height="24"
        sm="true"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm7.753 18.305c-.261-.586-.789-.991-1.871-1.241-2.293-.529-4.428-.993-3.393-2.945 3.145-5.942.833-9.119-2.489-9.119-3.388 0-5.644 3.299-2.489 9.119 1.066 1.964-1.148 2.427-3.393 2.945-1.084.25-1.608.658-1.867 1.246-1.405-1.723-2.251-3.919-2.251-6.31 0-5.514 4.486-10 10-10s10 4.486 10 10c0 2.389-.845 4.583-2.247 6.305z"
        />
      </svg>
      <p>
        Logout
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
